src/adapters/kafka/index.ts

```diff
@@ -0,0 +1,99 @@
+import { Kafka } from 'kafkajs'
+import Adapter from '../../lib/adapter.js'
+import GleeMessage from '../../lib/message.js'
+
+class KafkaAdapter extends Adapter {
+  private kafka: Kafka
+  private firstConnect: boolean = true
+  name(): string {
+    return 'Kafka adapter'
+  }
+
+  async connect() {
+    const kafkaOptions = await this.resolveProtocolConfig('kafka')
+    const securityRequirements = (this.AsyncAPIServer.security() || []).map(
+      (sec) => {
+        const secName = Object.keys(sec.json())[0]
+        return this.parsedAsyncAPI.components().securityScheme(secName)
+      }
+    )
+    const userAndPasswordSecurityReq = securityRequirements.find(
+      (sec) => sec.type() === 'userPassword'
+    )
+    const scramSha256SecurityReq = securityRequirements.find(
+      (sec) => sec.type() === 'scramSha256'
+    )
+    const scramSha512SecurityReq = securityRequirements.find(
+      (sec) => sec.type() === 'scramSha512'
+    )
+  
+    const brokerUrl = new URL(this.AsyncAPIServer.url())
+    this.kafka = new Kafka({
+      clientId: 'glee-app', 
+      brokers: [brokerUrl.host],
+      ssl: {
+        rejectUnauthorized: kafkaOptions?.authentication?.rejectUnauthorized,
+        key: kafkaOptions?.authentication?.keyFilePath,
```

Comment:
As pointed above, this should be `key` instead of `keyFilePath`:

```suggestion
        key: kafkaOptions?.authentication?.key,
```